doc/languages-frameworks/rust.section.md

```diff
@@ -912,6 +912,48 @@ You can try this out by:
 
 As of writing, this prints out `rustc 1.56.0 (09c42c458 2021-10-18)`.
 
+### How to use an overlay toolchain in a derivation  {#how-to-use-an-overlay-toolchain-in-a-derivation}
+
+You can also use an overlay's Rust toolchain with `buildRustPackage`.
+The below snippet demonstrates invoking `buildRustPackage` with an oxalica overlay selected Rust toolchain:
+```nix
+with import <nixpkgs>
+{
+  overlays = [
+    (import (fetchTarball "https://github.com/oxalica/rust-overlay/archive/master.tar.gz"))
+  ];
+};
+
+rustPlatform.buildRustPackage rec {
+  pname = "ripgrep";
+  version = "12.1.1";
+  nativeBuildInputs = [
+    rust-bin.stable.latest.minimal
```

Comment:
I'm not sure if that is the properway to do this. Did you test this ?
I've looked into buildRustPackage and there the rustc is passed as a nativeBuildInput, but rustc is passed at the end
https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/rust/default.nix#L115
I'm not sure this would actually work.
We need to test to make sure.

Reply:
> I'm not sure if that is the properway to do this. Did you test this ?

Good question. I tried smoke testing and noticed it did begin compiling but I did not verify the final result.

I am also having trouble testing now:
```
 nix-build default.nix 
/nix/store/p6hsrqpm4r8c7m1mks53lcn1p0i7mvpl-ripgrep-12.1.1
```
Given, these concerns I'd say to revert 2b6e40d and create a new issue to capture showing how to use an overlay in a derivation.

Could you share a snippet or a bit more context about showing how to use an overlay in a derivation? I was a bit unsure about that. Thanks!

Reply:
Sorry I wasn't super clear in my message.
Given the fact that rustc (in the snippet I've given) is passed after natviebuildInputs from rustBuildPackage, I think that the rust compiler used is always the default one. That's what I meant by this doesn't work. You could try testing this by adding a `postConfigure = "echo $(rustc --version)"` and checking the version given. I think it will give us the default rustc version used (1.55 at the moment) and not the one that you passed (latest stable is 1.56).
Let me know if it doesn't make sense. If you have time to test, I would be curious. I should have time to test myself next week in case you are busy.

Reply:
Actually, I just had time to test this on a project I was doing, and it is working like a charm. No more concerns from me.

Reply:
ah, I see what you are saying. Thanks for clarifying!

> Actually, I just had time to test this on a project I was doing, and it is working like a charm. No more concerns from me.

Oh, nice thanks! Curious, how did you test it? Wanted to compare with what I was doing for added learning 😄 .

Reply:
I took a project where I could use buildRustPackage, I added the overlay as a nativeBuildInput, then I added in the `postConfigure = "echo $(rustc --version)";` and I verified that it was 1.56 and not 1.55.